stdlib/LinearAlgebra/src/adjtrans.jl

```diff
@@ -107,56 +107,14 @@ IndexStyle(::Type{<:AdjOrTransAbsMat}) = IndexCartesian()
 
 
 # MemoryLayout of transposed and adjoint matrices
-abstract type ConjDenseColumns{T<:Complex} <: BlasMatrixLayout{T} end
-struct ConjDenseColumnMajor{T<:Complex} <: ConjDenseColumns{T} end
-struct ConjDenseColumnsStridedRows{T<:Complex} <: ConjDenseColumns{T} end
-abstract type ConjDenseRows{T<:Complex} <: BlasMatrixLayout{T} end
-struct ConjDenseRowMajor{T<:Complex} <: ConjDenseRows{T} end
-struct ConjDenseRowsStridedColumns{T<:Complex} <: ConjDenseRows{T} end
-struct ConjStridedLayout{T<:Complex} <: DenseLayout{T} end
-
-"""
-    ConjDenseColumnMajor{T}()
-
-is returned by `MemoryLayout(A)` if a vector or matrix `A` has storage in memory
-equivalent to the complex conjugate of an `Array`, so that `stride(A,1) == 1`
-and `stride(A,2) == size(A,1)`.
-`Array`s with `ConjDenseColumnMajor` must conform to the `DenseArray` interface
-and `adjoint(A)` should return a matrix whose layout is `DenseRowMajor{T}()`.
-"""
-ConjDenseColumnMajor
-
-"""
-    ConjDenseColumnsStridedRows{T}()
-
-is returned by `MemoryLayout(A)` if a vector or matrix `A` has storage in memory
-as a column major matrix, where the complex conjugates of the entries of `A`
-are stored.
-`Array`s with `ConjDenseColumnsStridedRows` must conform to the `DenseArray` interface,
-and `adjoint(A)` should return a matrix whose layout is `DenseRowsStridedColumns{T}()`.
-"""
-ConjDenseColumnsStridedRows
-
-"""
-    ConjDenseRowMajor{T}()
-
-is returned by `MemoryLayout(A)` if a vector or matrix `A` has storage in memory
-equivalent to the adjoint of an `Array`, so that `stride(A,1) == 1` and `stride(A,2) == size(A,1)`.
-`Array`s with `ConjDenseRowMajor` must conform to the `DenseArray` interface
-and `adjoint(A)` should return a matrix whose layout is `DenseColumnMajor{T}()`.
-"""
-ConjDenseRowMajor
-
-"""
-    ConjDenseRowsStridedColumns{T}()
+struct ConjLayout{T<:Complex, ML<:MemoryLayout} <: MemoryLayout{T}
+    layout::ML
+end
+ConjLayout(layout::MemoryLayout{T}) where T<:Complex = ConjLayout{T, typeof(layout)}(layout)
```

Comment:
With triangular dispatch, you can now define `ML<:MemoryLayout{T}` and the explicit constructor on line 113 would no longer be necessary (I think).